### PR TITLE
Fully merge WebArena

### PR DESCRIPTION
**Pull request by shuyanzhou**
This PR merges several independent developments from WebArena
- More robust in-viewport AXtree implementation. Note that the implementation will be slower since it makes client calls for every node in AXtree.
- Scripts to launch several scripts and evaluate them in parallel. The script also checks error runs and re-runs those examples
- Change the login to be per example to avoid unexpected logout.

### CODE REVIEW

File: browser_env/processors.py [hunk 1]
```diff
@@ -512,229 +596,199 @@ def dfs(idx: int, obs_node_id: str, depth: int) -> str:
...
-            tab_title_str = " | ".join(
-                ["Tab {idx}" for idx in range(len(open_tabs))]
-            )
+            tab_title_str = " | ".join([f"Tab {idx}" for idx in range(len(open_tabs))])
 
         try:
-            browser_info = self.fetch_browser_info(page, client)
+            browser_info = self.fetch_browser_info(page)
         except Exception:
             page.wait_for_load_state("load", timeout=500)
```
Comment (kohjingyu) [inline browser_env/processors.py:742]: Should this be `sleep_after_execution * 1000` instead of 500?

File: run.py [hunk 2]
```diff
@@ -493,7 +516,7 @@ def dump_config(args: argparse.Namespace) -> None:
     os.environ["TOKENIZERS_PARALLELISM"] = "false"
 
     args = config()
-    args.sleep_after_execution = 2.5
```
Comment (kohjingyu) [inline run.py:496]: Maybe revert this back to 2.5 to be safe?

**Review by kohjingyu — APPROVED** [review]
Some minor comments, LGTM otherwise. Thanks for doing this!

File: browser_env/processors.py (hunk 1 above)
Comment (shuyanzhou) [reply browser_env/processors.py:742]: to fetch browser info specifically, seems like 500ms is OK.

**Comment by shuyanzhou** [comment]
`gpt-3.5-turbo-16k-0613`  trajectories with `temperature = 0` and the 2-shot [no_na prompt](https://github.com/web-arena-x/visualwebarena/blob/main/agent/prompts/raw/p_cot_id_actree_2s_no_na.py) are [here](https://drive.google.com/file/d/1IhOEcqS3zSbHJBhLSOJoGFPJfM9wFSIR/view?usp=drive_link).

**Pull request merged**